app/models/prolong_token_wizard.rb

```diff
@@ -0,0 +1,42 @@
+class ProlongTokenWizard < ApplicationRecord
+  belongs_to :token
+
+  delegate :prolong!, to: :token
+
+  enum status: {
+    owner: 0,
+    project_purpose: 10,
+    contacts: 20,
+    summary: 30,
+    requires_update: 100,
+    prolonged: 110
+  }
+
+  validates :owner, presence: true, if: :owner_or_after?
+  validates :project_purpose, inclusion: { in: [true, false] }, if: :project_purpose_or_after?
+  validates :contact_technique, inclusion: { in: [true, false] }, if: :contacts_or_after?
+  validates :contact_metier, inclusion: { in: [true, false] }, if: :contacts_or_after?
```

Comment:
```suggestion
  validates :project_purpose, inclusion: { in: [true, false] }, if: -> { after?(:contacts) }
  validates :contact_technique, inclusion: { in: [true, false] }, if: -> { after?(:contacts) }
  validates :contact_metier, inclusion: { in: [true, false] }, if: -> { after?(:contacts) }
  
  def after?(kind)
    # Peut-être même juste `statuses` ici
    class.statuses[kind] >= status
  end
```

et tu flips les N truc en dessous là